check-vulnerabilities/check_vulnerabilities.py

```diff
@@ -0,0 +1,348 @@
+        raise RuntimeError(
+            "Safety results have not been generated... Something went wrong during",
+            "the execution of 'safety check -o bare --save-json info_safety.json'. ",
+            "Verify workflow logs.",
+        )
+
+    # Connect to the repository
+    g = github.Github(auth=github.Auth.Token(TOKEN))
+
+    # Get the repository
+    repo = g.get_repo(REPOSITORY)
+
+    # Get the available security advisories
+    existing_advisories = {}
+    pl_advisories = repo.get_repository_advisories()
+    for advisory in pl_advisories:
+        existing_advisories[advisory.summary] = advisory
+
+    ###############################################################################
+    # THIRD PARTY SECURITY ADVISORIES
+    ###############################################################################
+
+    # Process the detected advisories by Safety
+    safety_results_reported = 0
+    vulnerability: dict
+    for vulnerability in safety_results["vulnerabilities"]:
+        # Retrieve the needed values
+        v_id = vulnerability.get("vulnerability_id")
+        v_package = vulnerability.get("package_name")
+        v_cve = vulnerability.get("CVE")
+        v_url = vulnerability.get("more_info_url")
+        v_desc = vulnerability.get("advisory")
+        v_affected_versions = vulnerability.get("vulnerable_spec")
+        v_fixed_versions = vulnerability.get("fixed_versions")
+
+        # Advisory info
+        summary = f"Safety vulnerability {v_id} for package '{v_package}'"
+        vuln_adv = {
+            "package": {"name": f"{v_package}", "ecosystem": "pip"},
+            "vulnerable_version_range": f"{v_affected_versions}",
+            "patched_versions": f"{v_fixed_versions}",
+            "vulnerable_functions": [],
+        }
+        desc = f"""
+{v_desc}
+
+#### More information
+
+Visit {v_url} to find out more information.
+"""
+        # Check if the advisory already exists
+        if existing_advisories.get(summary):
+            continue
+        elif not DRY_RUN:
+            # New safety advisory detected
+            safety_results_reported += 1
+            new_advisory_detected = True
+
+            # Create the advisory but do not publish it
+            advisory = repo.create_repository_advisory(
+                summary=summary,
+                description=desc,
+                severity_or_cvss_vector_string="medium",
+                cve_id=v_cve,
+                vulnerabilities=[vuln_adv],
+            )
+
+            # Create an issue
+            if CREATE_ISSUES:
+                issue_body = f"""
+A new security advisory was open in this repository. See {advisory.html_url}.
+
+---
+**NOTE**
+
+Please update the security advisory status after evaluating. Publish the advisory
+once it has been verified (since it has been created in draft mode).
+
+---
+
+#### Description
+
+{desc}
+"""
+                repo.create_issue(title=summary, body=issue_body, labels=["security"])
```

Comment:
I wonder what happens here if you are using a token with no write permissions, could it the script fail here? I would prefer to check this earlier in the script rather than letting if fail mid script, but not sure how to do it anyway..

Reply:
This might be trickier. I need to check whether it is even possible

Reply:
Yeah this rings a bell... Then let's leave it as is